flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyedStateBackend.java

```diff
@@ -166,6 +167,13 @@ default boolean isSafeToReuseKVState() {
         return false;
     }
 
+    /**
+     * @return fixed lower-case string identifying the type of the underlying state backend, e.g.
+     *     rocksdb, hashmap, forst, batch, or unknown.
+     */
+    @Experimental
+    String getBackendTypeIdentifier();
```

Comment:
Shall we give default implementation returning 'unknown'?

Reply:
I don't have a strong preference. Rui suggested to have no default implementation so I removed it.
I didn't update the javadoc though - updated now